pkg/networking/route.go

```diff
@@ -0,0 +1,118 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/argoproj-labs/argocd-operator/pkg/argoutil"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RouteRequest objects contain all the required information to produce a route object in return
+type RouteRequest struct {
```

Comment:
@ciiay same for routes
please add `spec.Host`, 'spec.Port`, 'spec.Path`, `spec.TLS`, `spec.WildcardPolicy` etc